Public/Src/Engine/Scheduler/Scheduler.cs

```diff
@@ -5046,7 +5046,11 @@ private void PrioritizeAndSchedule(LoggingContext loggingContext, IEnumerable<No
                                 }
                             }
 
-                            pipRuntimeInfo.Priority = (criticalPath < 0 || criticalPath > MaxInitialPipPriority) ? MaxInitialPipPriority : unchecked((int)criticalPath);
+                            int priorityIncrements = (int.MaxValue / (Process.MaxPriority + 1));
+                            int priorityBase = m_pipTable.GetPipPriority(pipId) * priorityIncrements;
+                            int criticalPathPriority = (criticalPath < 0 || criticalPath > MaxInitialPipPriority) ? MaxInitialPipPriority : unchecked((int)criticalPath);
+                            criticalPathPriority = Math.Min(criticalPathPriority, priorityIncrements - 1);
+                            pipRuntimeInfo.Priority = priorityBase + criticalPathPriority;
 

```

Comment:
Some of us, like me, don't have the context of the work. So it's a bit hard to assess the work. Can you provide one?

Reply:
Added a description to the pr

---
In reply to: [280630658](https://github.com/Microsoft/BuildXL/pull/239#discussion_r280630658) [](ancestors = 280630658)

Reply:
Note to the person merging this pr. GitHub is different than vso. You'll have to do manual work to copy paste the or details into the merge commit as GitHub chose to append all commit messages in the or and totally ignore the pr info....

Reply:
good to know, thanks!

---
In reply to: [280865393](https://github.com/Microsoft/BuildXL/pull/239#discussion_r280865393) [](ancestors = 280865393)